mobile/App.js

```diff
@@ -6,25 +6,111 @@
  * @flow strict-local
  */
 
-import React, {useState} from 'react';
+import React, {useState, useRef, useEffect} from 'react';
 import {SafeAreaView} from 'react-native-safe-area-context';
+import AsyncStorage from '@react-native-async-storage/async-storage';
+
 import HomeView from './src/views/HomeView';
-import ScanView from './src/views/ScanView';
 import VendorView from './src/views/VendorView';
 import RegistrationView from './src/views/RegistrationView';
 import {NavigationContainer} from '@react-navigation/native';
 import {createBottomTabNavigator} from '@react-navigation/bottom-tabs';
 import ScreenOption from './src/utilites/ScreenOption';
-import {globalStyleSheet} from './src/utilites/Theme';
 import MyICardView from './src/views/MyICardView';
-
+import { AppState } from "react-native";
 import AuthContext from './src/context/AuthContext';
+import { storeDate } from './src/utilites/StoreDate';
+import { storeUser } from './src/utilites/StoreUser';
 
 const Tab = createBottomTabNavigator();
 
 const App = () => {
   const [user, setUser] = useState(null);
+  const [check, setCheck] = useState(false);
   const value = {user, setUser};
+  const appState = useRef(AppState.currentState);
+  
+  const logoutCheck = async(userData = null,startup = false) => {
+    
+    try{
+        const lastDate = parseInt(await AsyncStorage.getItem('lastOpened'));    
+        if(lastDate != null){
+            //change num in if statement below to a month in milliseconds
+            if(new Date().getTime() - lastDate > 2629800000){
+                setUser(null)
+                storeUser(null)
+            }
+            else{
+                if(startup){
+                  setUser(userData)
+                }
+                await storeDate()
+            }
+        }
+    } catch(error){
+        logoutCheck()
+    }
+}
+  useEffect(() => {
+    const checkUser = async() => {
+      const userData = JSON.parse(await AsyncStorage.getItem('userData'))
+      if(userData != null){
+        logoutCheck(userData, true)
+      }
+
+    } 
+    checkUser()
+  }, [])
+  
+  
+  useEffect(() => {
+    const subscription = AppState.addEventListener("change", nextAppState => {
+      
+      if (
+        appState.current.match(/active/) &&
+        nextAppState === "background"
+      ) {
+        
+        if(user != null){ 
+          storeDate()
+        }
+      }
+
+      appState.current = nextAppState;
+      
+    });
+
+    return () => {
+      subscription.remove();
+    };
+  }, [user]);
```

Comment:
If we are already storing date implicitly from the other event listener, we probably don't need this code to save it again when we close the app.

Reply:
I actually really did break my head trying to figure out how apostate worked lol. No matter what I did (even if it made sense completely), something didn't work. That's why I was hesitant to change anything even if some stuff is redundant coz I finally made it work

However, both event listeners are necessary. This is the one that captures the date whenever the user might be about to close the app (go to the background from foreground). The other event listener checks the date whenever the user comes back from background to foreground (the log out check). I think both are required

Reply:
I see what you mean, I was suggesting the idea that the time between when they open and close the app is negligible in the span of 30 days. 

If you do want to keep this code, you should merge it into the other useEffect hook since they both have the same dependencies